go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -135,7 +136,11 @@ func ToK8sPodSpec(ctx context.Context, tCtx pluginsCore.TaskExecutionContext) (*
 	return pod, nil
 }
 
-func BuildPodWithSpec(podTemplate *v1.PodTemplate, podSpec *v1.PodSpec) (*v1.Pod, error) {
+func PrimaryContainerNameMatches(containerName string, templateContainerName string, primaryContainerName string) bool {
```

Comment:
I don't think this necessarily needs to be a separate function. Probably, OK to just use inline.